Meta/CMake/utils.cmake

```diff
@@ -180,6 +180,19 @@ function(remove_path_if_version_changed version version_file cache_path)
     endif()
 endfunction()
 
+function(remove_cache_path_if_archive_sha256_mismatches archive cache_path sha256)
+    if (NOT EXISTS "${archive}")
+        return()
+    endif()
+
+    file(SHA256 "${archive}" actual_sha256)
```

Comment:
`file` runs at cmake time, right? So now we do O(n) sha256 computations every time we run cmake (with n being number of downloaded files).

It's be cooler if this were build edges that only ran if the inputs changed. I suppose that'd mean that the download would also happen at build time, hm.

Reply:
That's a fair point - yes, this runs the sha at configure time every time. Adds ~0.1 seconds on my machine (master is 2.45s, this branch is 2.55s).

I'll see what it looks like to do the temp file thing while downloading, then just sha-verify once after the download completes. I liked that this approach got rid of the version file shenanigans, but that's not a big deal

Reply:
That increase is about as long as gn runs in total for serenity 😬

Reply:
Alrighty, we now download to a .tmp file, and do SHA-256 verification only after downloading. Thanks!